maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/main/java/org/apache/maven/scm/provider/git/jgit/command/JGitUtils.java

```diff
@@ -461,5 +462,60 @@ public RevFilter clone()
             return revs;
         }
     }
+    
+    /**
+     * Remove all files in the given fileSet to the repository.
+     *
+     * @param git     the repo to remove the files from
+     * @param fileSet the set of files within the workspace, the files are removed
+     *                relative to the basedir of this fileset
+     * @return a list of removed files
+     * @throws GitAPIException
+     * @throws NoFilepatternException
+     */
+    public static List<ScmFile> removeAllFiles( Git git, ScmFileSet fileSet )
+        throws GitAPIException, NoFilepatternException
+    {
+        URI baseUri = fileSet.getBasedir().toURI();
+        RmCommand remove = git.rm();
+        for ( File file : fileSet.getFileList() )
+        {
+            if ( !file.isAbsolute() )
+            {
+                file = new File( fileSet.getBasedir().getPath(), file.getPath() );
+            }
+
+            if ( file.exists() )
+            {
+                String path = relativize( baseUri, file );
+                remove.addFilepattern( path );
+                remove.addFilepattern( file.getAbsolutePath() );
```

Comment:
Why is this added twice?

Reply:
@kwin Maybe you have an explanation for this...?

Reply:
RmCommand needs repository relative paths with forward slashes (https://archive.eclipse.org/jgit/site/5.0.1.201806211838-r/apidocs/org/eclipse/jgit/api/RmCommand.html#addFilepattern-java.lang.String-), therefore IMHO we need:

```
URI workingCopyRootUri = git.getRepository().getWorkTree().toURI();
String path = FilenameUtils.normalizeFilename( relativize( workingCopyRootUri, file ) );
remove.addFilepattern( path );
```